pkg/trace/stats/client_stats_aggregator.go

```diff
@@ -161,6 +166,23 @@ func (a *ClientStatsAggregator) flush(p []*pb.ClientStatsPayload) {
 	}
 }
 
+func (a *ClientStatsAggregator) setVersionDataFromContainerTags(p *pb.ClientStatsPayload) {
+	if p.ContainerID != "" {
+		gitCommitSha, imageTag, err := version.GetVersionDataFromContainerTags(p.ContainerID, a.conf)
+		if err != nil {
+			log.Error("Client stats aggregator is unable to resolve container ID (%s) to container tags: %v", p.ContainerID, err)
+		} else {
+			// Only override if the payload's original values were empty strings.
+			if p.ImageTag == "" {
+				p.ImageTag = imageTag
+			}
+			if p.GitCommitSha == "" {
+				p.GitCommitSha = gitCommitSha
+			}
```

Comment:
nit: If both of these fields are already filled out, there's no need to call the above function at all.